File: java/src/main/java/problems201_300/ContainsDuplicate.java
package problems201_300;

import java.util.Arrays;

enum ContainsDuplicate {
    DEFAULT {
        @Override
        boolean containsDuplicate(int[] nums) {
            Arrays.sort(nums);
            for (int i = 1; i < nums.length; i++) {
                if (nums[i] == nums[i - 1]) {
                    return true;
                }
            }
            return false;
        }
    };

    abstract boolean containsDuplicate(int[] nums);
}

File: kotlin/src/main/kotlin/problems201_300/ContainsDuplicate.kt
package problems201_300

import java.util.Arrays

enum class ContainsDuplicate {
    DEFAULT {
        override fun containsDuplicate(nums: IntArray): Boolean {
            Arrays.sort(nums)
            for (i in 1 until nums.size) {
                if (nums[i] == nums[i - 1]) {
                    return true
                }
            }
            return false
        }
    },
    ;

    abstract fun containsDuplicate(nums: IntArray): Boolean
}
